### PR TITLE
UPDATE iterating logic

### PR DESCRIPTION
**Pull request by RecuencoJones**
Hey again!

After digging a bit more into the logic for iterating globs and commands I noticed some issues with it:

- All commands would be ran for a file even if some previous command failed
- Not all files are awaited to be executed before exiting

I've changed it to use `Promise.all` and `for-of` so all executions are properly awaited even though executed in parallel, this seems to make `p-each-series` unnecessary.

Please, let me know what you think about it!

### CODE REVIEW

File: src/lint-changed.ts [hunk 1]
```diff
@@ -162,33 +161,31 @@ export async function lintChanged() {
...
+        }
+      } catch (e) {
+        error(e.message.trimLeft());
+        return e;
+      }
+    })));
+  }));
+
+  // In case there were any errors during execution, make sure to exit with error
+  if (results.some((command) => command.some((e) => e))) {
```
Comment (RecuencoJones) [inline src/lint-changed.ts:188]: maybe renaming `results` to `globResults` and `command` to `fileResults` will be easier to understand

```
globResults.some((fileResults) => fileResults.some((e) => e))
```

File: package.json [hunk 2]
```diff
@@ -32,7 +32,6 @@
     "await-to-js": "^2.1.1",
     "kleur": "^3.0.3",
     "micromatch": "^4.0.2",
-    "p-each-series": "^2.1.0",
```
Comment (damassi) [inline package.json:35]: Thank you!

**Review by damassi — APPROVED** [review]
💯

**Comment by artsyit** [comment]
:rocket: PR was released in [`v3.3.0`](https://github.com/artsy/lint-changed/releases/tag/v3.3.0) :rocket:

**Pull request merged**